### PR TITLE
PYIC-6959: Build test images

### PR DESCRIPTION
**Pull request by Wynndow**
### What changed

Build test images

### Why did it change

[PYIC-6959: Downgrade eslint](https://github.com/govuk-one-login/ipv-core-back/pull/2171/commits/d69bf456f8e5c1d17610162b2da43820a8461ec0)

Dependabot recently bumped eslint from 8 to 9, but this causes a
dependency resolution issue with typescript-eslint, which needs eslint
8.

[PYIC-6959: Build api-tests image](https://github.com/govuk-one-login/ipv-core-back/pull/2171/commits/c8b67d4d1f763c14968573479b073847781a0c81)

This adds Dockerfiles and a GHA workflow to build the api-tests images
for the pipeline.

It will only build the image if certain files have changed, to avoid
building it on every core-back merge, which would be unneccessary.

### Issue tracking

- [PYIC-6959](https://govukverify.atlassian.net/browse/PYIC-6959)

[PYIC-6959]: https://govukverify.atlassian.net/browse/PYIC-6959?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/secure-post-merge.yml [hunk 1]
```diff
@@ -11,7 +11,45 @@ on:
...
+      api-tests-changed: ${{ steps.get-changed-files.outputs.any_changed }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get changed api-tests files
+        id: get-changed-files
+        uses: tj-actions/changed-files@v44
```
Comment (Joe-Edwards-GDS) [inline .github/workflows/secure-post-merge.yml]: Sad that there's no out of the box solution for this :(

A cursory google also reveals https://github.com/dorny/paths-filter which appears to be a bit more focused on this use-case, but I don't have a clear idea which is best.

Might be worth flagging to @huwd and perhaps @TobySaundersGDS that we'd like to introduce a 3rd-party action here. There's a formal process for npm dependencies, but I'm not sure about GHAs (https://govukverify.atlassian.net/wiki/spaces/DIWAY/pages/4335697997/Library+approval+process)

I do think it's quite valuable to add this though!

**Review by Joe-Edwards-GDS — COMMENTED** [review]
This looks reasonable to me, but might be worth asking for @TobySaundersGDS (or someone else from dev-platform) for a review as well

File: .github/workflows/secure-post-merge.yml (hunk 1 above)
Comment (Wynndow) [reply .github/workflows/secure-post-merge.yml]: Have pinged them a message, see what they think. The two actions look pretty similar in functionality, just the syntax seems marginally different. Both look like well used and active repo's. ¯\_(ツ)_/¯

Comment (Wynndow) [reply .github/workflows/secure-post-merge.yml]: @Joe-Edwards-GDS Have checked and it's ok - @TobySaundersGDS suggested pinning to a particular SHA though, which I've done.

**Review by Joe-Edwards-GDS — APPROVED** [review]
Looks plausible - obviously worth keeping an eye on when we merge, as I'm sure you will :)

**Pull request merged**